rls/src/actions/format.rs

```diff
@@ -193,7 +193,8 @@ fn random_file() -> Result<(File, PathBuf), Error> {
 
 fn gen_config_file(config: &Config) -> Result<(File, PathBuf), Error> {
     let (mut file, path) = random_file()?;
-    let toml = config.all_options().to_toml().map_err(Error::ConfigTomlOutput)?;
+    let toml =
+        config.all_options().to_toml().map_err(|e| Error::ConfigTomlOutput(e.to_string()))?;
```

Comment:
@topecongiro - this means that we would technically have a breaking API change in rustfmt 1.4.18 on `to_toml()` (the error value is now `ToTomlError` instead of `String`) due to the backporting of https://github.com/rust-lang/rustfmt/pull/3948. I actually forgot that this was public in the rustfmt lib and consumed by rls. 

I've opened this as-is in case you and @Xanewok want to accept this change and proceed in order to fix the broken toolstates. Otherwise we'd probably want to yank v.14.18 of rustfmt, and then either publish v1.4.19 that still has the parser bug users keep reporting, or try to reimplement a fix using the now-deprecated `failure`